### ISSUE TITLE
px crash if ps wrapped in shell script

### ISSUE THREAD

**Issue by lesmana** [walles/px#24]
when I run `px` i get this:

```
$ px
Traceback (most recent call last):
  File "/home/lesmana/opt/px/bin/px", line 11, in <module>
    sys.exit(main())
  File "/home/lesmana/opt/px/lib/python3.6/site-packages/px/px.py", line 92, in main
    procs = px_process.get_all()
  File "/home/lesmana/opt/px/lib/python3.6/site-packages/px/px_process.py", line 262, in get_all
    resolve_links(all, now)
  File "/home/lesmana/opt/px/lib/python3.6/site-packages/px/px_process.py", line 244, in resolve_links
    process.parent = pid2process[process.ppid]
KeyError: 18698
```

The number (in this case `18698`) changes with each invocation.

I debugged through the code and found the problem: I have my `ps` wrapped in a shell script. `px` removes itself and all children from the process list. That leaves the real `ps` process (a grandchild in my case) as an orphan without parent.

Here is the self and child removal: https://github.com/walles/px/blob/2d683161879d6264cb67fade3a61b9b3251411a4/px/px_process.py#L248

The fix would be to remove self and all descendants. I have fixed this: https://github.com/lesmana/px/commit/e52c7cd7a5b9741fcdc0e54e3317d7962fb6c25b

I will submit a pull request shortly.

**Comment by walles** [comment]
Just out of curiosity, why do you have `ps` wrapped in a shell script? What service does the shell script provide?

**Comment by lesmana** [comment]
this is my wrapper scipt

```
#! /bin/sh

if [ $# -eq 0 ] ; then
  /usr/bin/ps -u "$USER" -o start_time,tty,ppid,pid,command --sort start_time
else
  /usr/bin/ps "$@"
fi
```
basically it changes the default behaviour of ps.

i have quite a few such wrapper scripts in my `~/bin`. some to change default behaviour. some to prepare environment before running. some to run a custom compiled software instead of the system installed one.

**Comment by walles** [comment]
Closed by #28
